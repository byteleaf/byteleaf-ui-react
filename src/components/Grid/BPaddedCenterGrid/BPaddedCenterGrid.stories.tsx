import React from 'react';
import { Button } from '@material-ui/core';

import { BPaddedCenterGrid } from './BPaddedCenterGrid';

export default {
  title: 'Grid|BPaddedCenterGrid',
};

export const BPaddedCenterGridStory = () => (
  <BPaddedCenterGrid>
    <Button>Submit</Button>
  </BPaddedCenterGrid>
);

BPaddedCenterGridStory.story = {
  name: 'default',
  parameters: {
    notes: 'A padded centered grid item',
  },
};
