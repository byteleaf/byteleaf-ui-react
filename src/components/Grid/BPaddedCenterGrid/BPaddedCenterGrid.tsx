import { Grid, styled, Theme } from '@material-ui/core';
import React from 'react';

export const BPaddedCenterGrid = styled(Grid)(({ theme }: { theme: Theme }) => ({
  paddingTop: theme.spacing(2),
  textAlign: 'center',
}));
