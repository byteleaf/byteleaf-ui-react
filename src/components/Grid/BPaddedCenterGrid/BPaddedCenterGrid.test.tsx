import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { BPaddedCenterGrid } from './BPaddedCenterGrid';

describe('BPaddedCenterGrid', () => {
  it('renders correctly', async () => {
    const { container } = render(
      <BPaddedCenterGrid>
        <button>test</button>
      </BPaddedCenterGrid>,
    );

    expect(container).toMatchSnapshot();
  });
});
