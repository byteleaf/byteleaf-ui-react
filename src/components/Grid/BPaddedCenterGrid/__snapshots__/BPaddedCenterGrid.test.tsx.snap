// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BPaddedCenterGrid renders correctly 1`] = `
<div>
  <div
    class="MuiGrid-root WithStyles(ForwardRef(Grid))-root-1 WithStyles(ForwardRef(Grid))-root-2"
  >
    <button>
      test
    </button>
  </div>
</div>
`;
